text/0000-cargo-install.md

```diff
@@ -0,0 +1,289 @@
+environment variable. This will be recommended when `cargo install` is run if
+`PATH` does not already look like it's configured.
+
+#### Crate Sources
+
+The `cargo install` command will be able to install crates from any source that
+Cargo already understands. For example it will start off being able to install
+from crates.io, git repositories, and local paths. Like with normal
+dependencies, downloads from crates.io can specify a version, git repositories
+can specify branches, tags, or revisions.
+
+#### Sources with multiple crates
+
+Sources like git repositories and paths can have multiple crates inside them,
+and Cargo needs a way to figure out which one is being installed. If there is
+more than one crate in a repo (or path), then Cargo will apply the following
+heuristics to select a crate, in order:
+
+1. If the `-p` argument is specified, use that crate.
+2. If only one crate has binaries, use that crate.
+3. If only one crate has examples, use that crate.
+4. Print an error suggesting the `-p` flag.
+
+#### Multiple binaries in a crate
+
+Once a crate has been selected, Cargo will by default build all binaries and
+install them. This behavior can be modified with the `--bin` or `--example`
+flags to configure what's installed on the local system.
+
+#### Building a Binary
+
+The `cargo install` command has some standard build options found on `cargo
+build` and friends, but a key difference is that `--release` is the default for
+installed binaries so a `--debug` flag is present to switch this back to
+debug-mode. Otherwise the `--features` flag can be specified to activate various
+features of the crate being installed.
+
+The `--target` option is omitted as `cargo install` is not intended for creating
+cross-compiled binaries to ship to other platforms.
+
+#### Conflicting Crates
+
+Cargo will not namespace the installation directory for crates, so conflicts may
+arise in terms of binary names. For example if crates A and B both provide a
+binary called `foo` they cannot be both installed at once. Cargo will reject
+these situations and recommend that a binary is selected via `--bin` or the
+conflicting crate is uninstalled.
+
+#### Placing output artifacts
+
+The `cargo install` command can be customized where it puts its output artifacts
+to install packages in a custom location. The root directory of the installation
+will be determined in a hierarchical fashion, choosing the first of the
+following that is specified:
+
+1. The `--root` argument on the command line.
+2. The environment variable `CARGO_INSTALL_ROOT`.
+3. The `install.root` configuration option.
+4. The value of `$CARGO_HOME` (also determined in an independent and
+   hierarchical fashion).
+
+Once the root directory is found, Cargo will place all binaries in the
+`$INSTALL_ROOT/bin` folder. Cargo will also reserve the right to retain some
+metadata in this folder in order to keep track of what's installed and what
+binaries belong to which package.
```

Comment:
This should probably be prefix to be consistent. That way, one could specify `cargo install --prefix /usr` to install system-wide binaries.